src/partition/src/row_splitter.rs

```diff
@@ -0,0 +1,322 @@
+                semantic_type: SemanticType::Tag as i32,
+            },
+            ColumnSchema {
+                column_name: "age".to_string(),
+                datatype: ColumnDataType::Uint32 as i32,
+                semantic_type: SemanticType::Field as i32,
+            },
+        ];
+        let rows = vec![
+            Row {
+                values: vec![
+                    ValueData::StringValue("1".to_string()).into(),
+                    ValueData::StringValue("Smith".to_string()).into(),
+                    ValueData::U32Value(20).into(),
+                ],
+            },
+            Row {
+                values: vec![
+                    ValueData::StringValue("2".to_string()).into(),
+                    ValueData::StringValue("Johnson".to_string()).into(),
+                    ValueData::U32Value(21).into(),
+                ],
+            },
+            Row {
+                values: vec![
+                    ValueData::StringValue("3".to_string()).into(),
+                    ValueData::StringValue("Williams".to_string()).into(),
+                    ValueData::U32Value(22).into(),
+                ],
+            },
+        ];
+        RowInsertRequest {
+            table_name: "t".to_string(),
+            rows: Some(Rows { schema, rows }),
+            region_number: 0,
+        }
+    }
+
+    #[derive(Debug, Serialize, Deserialize)]
+    struct MockPartitionRule;
+
+    impl PartitionRule for MockPartitionRule {
+        fn as_any(&self) -> &dyn Any {
+            self
+        }
+
+        fn partition_columns(&self) -> Vec<String> {
+            vec!["id".to_string()]
+        }
+
+        fn find_region(&self, values: &[Value]) -> Result<RegionNumber> {
+            let val = values.get(0).unwrap().clone();
+            let val = match val {
+                Value::String(v) => v.as_utf8().to_string(),
+                _ => unreachable!(),
+            };
+
+            Ok(val.parse::<u32>().unwrap() % 2)
+        }
+
+        fn find_regions_by_exprs(&self, _: &[PartitionExpr]) -> Result<Vec<RegionNumber>> {
+            unimplemented!()
+        }
+    }
+
+    #[derive(Debug, Serialize, Deserialize)]
+    struct MockMissedColPartitionRule;
+
+    impl PartitionRule for MockMissedColPartitionRule {
+        fn as_any(&self) -> &dyn Any {
+            self
+        }
+
+        fn partition_columns(&self) -> Vec<String> {
+            vec!["missed_col".to_string()]
+        }
+
+        fn find_region(&self, values: &[Value]) -> Result<RegionNumber> {
+            let val = values.get(0).unwrap().clone();
+            let val = match val {
+                Value::Null => 1,
+                _ => 0,
+            };
+
+            Ok(val)
+        }
+
+        fn find_regions_by_exprs(&self, _: &[PartitionExpr]) -> Result<Vec<RegionNumber>> {
+            unimplemented!()
+        }
+    }
+
+    #[derive(Debug, Serialize, Deserialize)]
+    struct EmptyPartitionRule;
```

Comment:
non-blocking: Could we reuse structs like `EmptyPartitionRule`, `MockMissedColPartitionRule`and `MockPartitionRule`  in tests?

Reply:
Not over yet, stay tuned for the next PR.